File: Week04/Day-02/garden/tree.ts
import { Plant } from "./plant";

'use strict';

export class Tree extends Plant {
  constructor(color: string, type: string = 'tree', waterLevel: number = 0, needWater: boolean = true, waterAbsorb: number = 0.4) {
    super(color, type, waterLevel, needWater, waterAbsorb);
    if (color === 'purple' || color === 'orange') {
      this.color = color;
    } else {
      throw new Error ('You can only choose purple or orange flowers buddy.');
    }
    if (this.waterLevel < 10) {
      this.needWater = true;
    } else {
      this.needWater = false;
    }
  } 
}

File: Week10/garden-app/tree.py
from plant import Plant

class Tree(Plant):

    def __init__(self, color):
        super().__init__(color)
        self.water_absorb = 0.4

        if self.water_level < 10:
            self.need_water = True
        else:
            self.need_water = False

tree1 = Tree('orange')
print(tree1.water_level)